server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -641,6 +644,31 @@ public int resolveTrackTotalHitsUpTo() {
         return resolveTrackTotalHitsUpTo(scroll, source);
     }
 
+    @Override
+    public SearchRequest rewrite(QueryRewriteContext ctx) throws IOException {
+        if (source == null) {
+            return this;
+        }
+
+        SearchSourceBuilder source = this.source.rewrite(ctx);
+        boolean hasChanged = source != this.source;
+
+        // add a sort tiebreaker for PIT search requests if not explicitly set
+        if (source.pointInTimeBuilder() != null) {
+            if (source.sorts() == null || source.sorts().isEmpty()) {
+                source.sort(SortBuilders.scoreSort());
+            }
```

Comment:
I am not super clear why do we need to add sort on `_shard_doc` if the initial query was without sort at all?

Reply:
synchronized with @jimczi offline, and indeed we intend to add `_shard_doc` even for queries without any explicit sort. As these queries implicitly use sort on _score + shard + _doc. With this PR we make this sort obvious.